File: .config/i3/lemonbar/terminus-font-4.46/bin/fnutil.py
#
# Copyright (c) 2017 Dimitar Toshkov Zhekov <dimitar.zhekov@gmail.com>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation; either version 2 of
# the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#

import sys

def parseDec(name, s, minValue, maxValue):
	value = int(s)

	if value < minValue:
		raise Exception('{0} must be >= {1}'.format(name, minValue))

	if value > maxValue:
		raise Exception('{0} must be <= {1}'.format(name, maxValue))

	return value


def parseHex(name, s, minValue=0, maxValue=0x10FFFF):
	value = int(s, 16)

	if value < minValue:
		raise Exception('{0} must be >= {1}'.format(name, hex(minValue)[2:].upper()))

	if value > maxValue:
		raise Exception('{0} must be <= {1}'.format(name, hex(maxValue)[2:].upper()))

	return value


def unQuote(s, name=None):
	if len(s) >= 2 and s.startswith('"') and s.endswith('"'):
		s = s[1:len(s) - 1]
	elif name != None:
		raise Exception(name + ' must be quoted')

	return s


def startCli(programName, parseArgv, argopts, mainProgram):
	excstk = False

	try:
		if sys.hexversion < 0x3040000:
			raise Exception('python 3.4.0 or later required')

		# should replace the global option values with a dict
		optind = 1
		while optind < len(sys.argv):
			opt = sys.argv[optind]

			if opt == '-' or not opt.startswith('-'):
				break

			if opt == '--':
				optind += 1
				break

			if opt == '--excstk':
				excstk = True
			else:
				optarg = None

				if argopts.find(opt[1]) != -1:
					if len(opt) == 2:
						optind += 1
						if optind == len(sys.argv):
							raise Exception('option ' + opt + ' requires an argument')

						optarg = sys.argv[optind]
					else:
						optarg = opt[2:]
						opt = opt[0:2]

				parseArgv(opt, optarg)

			optind += 1

		mainProgram(optind)
	except Exception as e:
		if excstk:
			raise
		else:
			sys.stderr.write('{0}: {1}\n'.format(programName, e))
			sys.exit(1)


def showLicense(prefix):
	sys.stdout.write(prefix +
		'\n' +
		'This program is free software; you can redistribute it and/or\n' +
		'modify it under the terms of the GNU General Public License as\n' +
		'published by the Free Software Foundation; either version 2 of\n' +
		'the License, or (at your option) any later version.\n' +
		'\n' +
		'This program is distributed in the hope that it will be useful,\n' +
		'but WITHOUT ANY WARRANTY; without even the implied warranty of\n' +
		'MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the\n' +
		'GNU General Public License for more details.\n')

File: .config/i3/lemonbar/terminus-font-4.46/bin/fnutil.js
//
// Copyright (c) 2017 Dimitar Toshkov Zhekov <dimitar.zhekov@gmail.com>
//
// This program is free software; you can redistribute it and/or
// modify it under the terms of the GNU General Public License as
// published by the Free Software Foundation; either version 2 of
// the License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//

'use strict';

const util = require('util');

function parseDec(name, s, minValue, maxValue) {
	let value = parseInt(s, 10);

	if (s.match(/^-?\d+$/) === null) {
		throw new Error(util.format('invalid %s format', name));
	}
	if (value < minValue) {
		throw new Error(util.format('%s must be >= %d', name, minValue));
	}
	if (value > maxValue) {
		throw new Error(util.format('%s must be <= %d', name, maxValue));
	}

	return value;
}

function parseHex(name, s, minValue, maxValue) {
	let value = parseInt(s, 16);

	if (s.match(/^[\dA-Fa-f]+$/) === null) {
		throw new Error(util.format('invalid %s format', name));
	}
	if (minValue == null) {
		minValue = 0;
	}
	if (maxValue == null) {
		maxValue = 0x10FFFF;
	}
	if (value < minValue) {
		throw new Error(util.format('%s must be >= %s', minValue.toString(16).toUpperCase()));
	}
	if (value > maxValue) {
		throw new Error(util.format('%s must be <= %s', maxValue.toString(16).toUpperCase()));
	}

	return value;
}

function unQuote(s, name) {
	if (s.length >= 2 && s.startsWith('"') && s.endsWith('"')) {
		s = s.substring(1, s.length - 1);
	} else if (name != null) {
		throw new Error(name + ' must be quoted');
	}

	return s;
}

function startCli(programName, parseArgv, argopts, mainProgram) {
	let excstk = false;

	try {
		let optind;

		if (Number(process.version.match(/^v?(\d+\.\d+)/)[1]) < 6.8) {
			throw new Error('node v6.8.0 or later required');
		}

		// should replace the global option values with a dict
		for (optind = 2; optind < process.argv.length; optind++) {
			let opt = process.argv[optind];

			if (opt === '-' || !opt.startsWith('-')) {
				break;
			}

			if (opt === '--') {
				optind++;
				break;
			}

			if (opt === '--excstk') {
				excstk = true;
			} else {
				let optarg = null;

				if (argopts.indexOf(opt[1]) !== -1) {
					if (opt.length === 2) {
						if (++optind === process.argv.length) {
							throw new Error('option ' + opt + ' requires an argument');
						}
						optarg = process.argv[optind];
					} else {
						optarg = opt.substring(2);
						opt = opt.substring(0, 2);
					}
				}

				parseArgv(opt, optarg);
			}
		}

		mainProgram(optind);
	} catch (e) {
		if (excstk) {
			throw e;
		} else {
			process.stderr.write(util.format('%s: %s\n', programName, e.message));
			process.exit(1);
		}
	}
}

function showLicense(prefix) {
	process.stdout.write(prefix +
		'\n' +
		'This program is free software; you can redistribute it and/or\n' +
		'modify it under the terms of the GNU General Public License as\n' +
		'published by the Free Software Foundation; either version 2 of\n' +
		'the License, or (at your option) any later version.\n' +
		'\n' +
		'This program is distributed in the hope that it will be useful,\n' +
		'but WITHOUT ANY WARRANTY; without even the implied warranty of\n' +
		'MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the\n' +
		'GNU General Public License for more details.\n'
	);
}

// EXPORTS
module.exports = Object.freeze({
	parseDec,
	parseHex,
	unQuote,
	startCli,
	showLicense
});
